internal/controller/utils.go

```diff
@@ -522,3 +547,123 @@ func healthCheck(sts *appsv1.StatefulSet, lg klog.Logger) (*clientv3.MemberListR
 
 	return memberlistResp, healthInfos, nil
 }
+
+func createCMCertificateConfig(ec *ecv1alpha1.ProviderCertManagerConfig) *certInterface.Config {
+	duration, err := time.ParseDuration(ec.ValidityDuration)
+	if err != nil {
+		log.Printf("Failed to parse ValidityDuration: %s", err)
+	}
+	config := &certInterface.Config{
+		CommonName:       ec.CommonName,
+		Organization:     ec.Organization,
+		ValidityDuration: duration,
+		AltNames: certInterface.AltNames{
+			DNSNames: ec.AltNames.DNSNames,
+			IPs:      make([]net.IP, len(ec.AltNames.DNSNames)),
+		},
+		ExtraConfig: map[string]any{
+			"issuerName": ec.IssuerName,
+			"issuerKind": ec.IssuerKind,
+		},
+	}
+	return config
+}
+
+func createAutoCertificateConfig(ec *ecv1alpha1.ProviderAutoConfig) *certInterface.Config {
+	// TODO
+	config := &certInterface.Config{}
+	return config
+}
+
+func createCertificate(ec *ecv1alpha1.EtcdCluster, ctx context.Context, c client.Client, certName string) error {
+	cert, certErr := certificate.NewProvider(certificate.ProviderType(ec.Spec.TLS.Provider), c)
+	if certErr != nil {
+		// TODO: instead of error, set default autoConfig
+		return certErr
+	}
+	_, getCertError := cert.GetCertificateConfig(ctx, certName, ec.Namespace)
+	if getCertError != nil {
+		if k8serrors.IsNotFound(getCertError) {
+			log.Printf("Creating certificate: %s for etcd-operator: %s\n", certName, ec.Name)
+			switch {
+			case ec.Spec.TLS.ProviderCfg.AutoCfg != nil:
+				cmConfig := createAutoCertificateConfig(ec.Spec.TLS.ProviderCfg.AutoCfg)
+				createCertErr := cert.EnsureCertificateSecret(ctx, certName, ec.Namespace, cmConfig)
+				if createCertErr != nil {
+					log.Printf("Error creating certificate: %s", createCertErr)
+				}
+				return nil
+			case ec.Spec.TLS.ProviderCfg.CertManagerCfg != nil:
+				cmConfig := createCMCertificateConfig(ec.Spec.TLS.ProviderCfg.CertManagerCfg)
+				createCertErr := cert.EnsureCertificateSecret(ctx, certName, ec.Namespace, cmConfig)
+				if createCertErr != nil {
+					log.Printf("Error creating certificate: %s", createCertErr)
+				}
+				return nil
+			default:
+				if ec.Spec.TLS.ProviderCfg.AutoCfg == nil {
+					// TODO: instead of error, set default autoConfig which will be applied if Provider/ProviderConfig is not set
+					return errors.New("default autoCertificate config not defined")
+				}
```

Comment:
If etcd-operator runs into this branch, it means both `AutoCfg` and `CertMangerCfg` are `nil`. In this case, we use AuthProvider.